internal/config/conmonmgr/conmonmgr.go

```diff
@@ -0,0 +1,107 @@
+package conmonmgr
+
+import (
+	"path"
+	"regexp"
+	"strconv"
+
+	"github.com/cri-o/cri-o/utils/cmdrunner"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	majorVersionSupportsSync = 2
+	minorVersionSupportsSync = 0
+	patchVersionSupportsSync = 19
+)
+
+type ConmonManager struct {
+	majorVersion int
+	minorVersion int
+	patchVersion int
+
+	supportsSync bool
+}
+
+// this function is heavily based on github.com/containers/common#probeConmon
+func New(conmonPath string) (*ConmonManager, error) {
+	return newWithCommandRunner(conmonPath, &cmdrunner.RealCommandRunner{})
+}
+
+func newWithCommandRunner(conmonPath string, runner cmdrunner.CommandRunner) (*ConmonManager, error) {
+	if !path.IsAbs(conmonPath) {
+		return nil, errors.Errorf("conmon path is not absolute: %s", conmonPath)
+	}
+	out, err := runner.CombinedOutput(conmonPath, "--version")
+	if err != nil {
+		return nil, err
```

Comment:
Let's wrap those errors to provide a more detailed output:
```suggestion
		return nil, errors.Wrap(err, "get conmon version")
```